### PR TITLE
start a `dev checklist`

### PR DESCRIPTION
**Pull request by jcupitt**
How about a "Developer checklist" chapter in the docs listing things you (ideally) should / should not do? It could gather up many common questions and issues we see many times in discussions.

I put in a few obvious ones, but we should add more. Any thoughts?

### CODE REVIEW

File: doc/Developer-checklist.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+libvips is a slightly unusual library and you need to consider some features
+when you design software that uses it.
+
+## If you can, use `thumbnail`, not `resize`
+
+The `thumbnail` operation combines load and resize into one step. This lets it
+take advantage of format library features, such as shrink on load, and can
+lead to a large improvement in speed and drop in memory use.
+
+For example, whith this JPEG image:
```
Comment (kleisauke) [inline doc/Developer-checklist.md:21]:
```suggestion
For example, with this JPEG image:
```

File: doc/Developer-checklist.xml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+<para>
+  libvips is a slightly unusual library and you need to consider some features when you design software that uses it.
+</para>
+<refsect3 xml:id="if-you-can-use-thumbnail-not-resize">
+  <title>If you can, use <literal>thumbnail</literal>, not <literal>resize</literal></title>
+  <para>
+    The <literal>thumbnail</literal> operation combines load and resize into one step. This lets it take advantage of format library features, such as shrink on load, and can lead to a large improvement in speed and drop in memory use.
+  </para>
+  <para>
+    For example, whith this JPEG image:
```
Comment (kleisauke) [inline doc/Developer-checklist.xml:22]:
```suggestion
    For example, with this JPEG image:
```

**Review by kleisauke — APPROVED** [review]
Great idea, LGTM!

**Comment by jcupitt** [comment]
Thanks!

**Pull request merged**